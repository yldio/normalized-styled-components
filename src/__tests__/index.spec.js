import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import {
  Article,
  Aside,
  Footer,
  Header,
  Nav,
  Section,
  FigCaption,
  Figure,
  Main,
  H1,
  Hr,
  Pre,
  A,
  Abbr,
  B,
  Strong,
  Code,
  Kbd,
  Samp,
  Dfn,
  Mark,
  Small,
  Sub,
  Sup,
  Audio,
  Video,
  Img,
  Svg,
  Button,
  Input,
  Optgroup,
  Select,
  Textarea,
  Fieldset,
  Legend,
  Progress,
  Details,
  Menu,
  Summary,
  Canvas,
} from '../';

it('renders <Article /> correctly', () => {
  const tree = renderer.create(<Article />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Aside /> correctly', () => {
  const tree = renderer.create(<Aside />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Footer /> correctly', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Header /> correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Nav /> correctly', () => {
  const tree = renderer.create(<Nav />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Section /> correctly', () => {
  const tree = renderer.create(<Section />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <FigCaption /> correctly', () => {
  const tree = renderer.create(<FigCaption />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Figure /> correctly', () => {
  const tree = renderer.create(<Figure />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Main /> correctly', () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <H1 /> correctly', () => {
  const tree = renderer.create(<H1 />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Hr /> correctly', () => {
  const tree = renderer.create(<Hr />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Pre /> correctly', () => {
  const tree = renderer.create(<Pre />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <A /> correctly', () => {
  const tree = renderer.create(<A />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Abbr /> correctly', () => {
  const tree = renderer.create(<Abbr />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <B /> correctly', () => {
  const tree = renderer.create(<B />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Strong /> correctly', () => {
  const tree = renderer.create(<Strong />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Code /> correctly', () => {
  const tree = renderer.create(<Code />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Kbd /> correctly', () => {
  const tree = renderer.create(<Kbd />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Samp /> correctly', () => {
  const tree = renderer.create(<Samp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Dfn /> correctly', () => {
  const tree = renderer.create(<Dfn />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Mark /> correctly', () => {
  const tree = renderer.create(<Mark />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Small /> correctly', () => {
  const tree = renderer.create(<Small />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Sub /> correctly', () => {
  const tree = renderer.create(<Sub />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Sup /> correctly', () => {
  const tree = renderer.create(<Sup />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Audio /> correctly', () => {
  const tree = renderer.create(<Audio />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Video /> correctly', () => {
  const tree = renderer.create(<Video />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Img /> correctly', () => {
  const tree = renderer.create(<Img />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Svg /> correctly', () => {
  const tree = renderer.create(<Svg />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Button /> correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Input /> correctly', () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Optgroup /> correctly', () => {
  const tree = renderer.create(<Optgroup />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Select /> correctly', () => {
  const tree = renderer.create(<Select />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Textarea /> correctly', () => {
  const tree = renderer.create(<Textarea />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Fieldset /> correctly', () => {
  const tree = renderer.create(<Fieldset />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Legend /> correctly', () => {
  const tree = renderer.create(<Legend />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Progress /> correctly', () => {
  const tree = renderer.create(<Progress />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Details /> correctly', () => {
  const tree = renderer.create(<Details />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Menu /> correctly', () => {
  const tree = renderer.create(<Menu />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Summary /> correctly', () => {
  const tree = renderer.create(<Summary />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <Canvas /> correctly', () => {
  const tree = renderer.create(<Canvas />).toJSON();
  expect(tree).toMatchSnapshot();
});
