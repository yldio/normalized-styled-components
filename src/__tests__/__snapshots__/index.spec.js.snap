// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <A /> correctly 1`] = `
.c0 {
  background-color: transparent;
  -webkit-text-decoration-skip: objects;
  text-decoration-skip: objects;
}

<a
  className="c0"
/>
`;

exports[`renders <Abbr /> correctly 1`] = `
.c0[title] {
  border-bottom: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration: underline dotted;
  text-decoration: underline dotted;
}

<abbr
  className="c0"
/>
`;

exports[`renders <Article /> correctly 1`] = `
.c0 {
  display: block;
}

<article
  className="c0"
/>
`;

exports[`renders <Aside /> correctly 1`] = `
.c0 {
  display: block;
}

<aside
  className="c0"
/>
`;

exports[`renders <Audio /> correctly 1`] = `
.c0 {
  display: inline-block;
}

.c0:not([controls]) {
  display: none;
  height: 0;
}

<audio
  className="c0"
/>
`;

exports[`renders <B /> correctly 1`] = `
.c0 {
  font-weight: bolder;
}

<b
  className="c0"
/>
`;

exports[`renders <Button /> correctly 1`] = `
.c0 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
}

.c0::-moz-focus-inner,
.c0[type='button']::-moz-focus-inner,
.c0[type='reset']::-moz-focus-inner,
.c0[type='submit']::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c0:-moz-focusring,
.c0[type='button']:-moz-focusring,
.c0[type='reset']:-moz-focusring,
.c0[type='submit']:-moz-focusring {
  outline: 0.0625rem dotted ButtonText;
}

<button
  className="c0"
/>
`;

exports[`renders <Canvas /> correctly 1`] = `
.c0 {
  display: inline-block;
}

<canvas
  className="c0"
/>
`;

exports[`renders <Code /> correctly 1`] = `
.c0 {
  font-family: monospace,monospace;
  font-size: 1rem;
}

<code
  className="c0"
/>
`;

exports[`renders <Details /> correctly 1`] = `
.c0 {
  display: block;
}

<details
  className="c0"
/>
`;

exports[`renders <Dfn /> correctly 1`] = `
.c0 {
  font-style: italic;
}

<dfn
  className="c0"
/>
`;

exports[`renders <Fieldset /> correctly 1`] = `
.c0 {
  padding: 0.35em 0.75em 0.625em;
}

<fieldset
  className="c0"
/>
`;

exports[`renders <FigCaption /> correctly 1`] = `
.c0 {
  display: block;
}

<figcaption
  className="c0"
/>
`;

exports[`renders <Figure /> correctly 1`] = `
.c0 {
  display: block;
  margin: 1rem 2.5rem;
}

<figure
  className="c0"
/>
`;

exports[`renders <Footer /> correctly 1`] = `
.c0 {
  display: block;
}

<footer
  className="c0"
/>
`;

exports[`renders <H1 /> correctly 1`] = `
.c0 {
  font-size: 2rem;
  margin: 0.625rem 0;
}

<h1
  className="c0"
/>
`;

exports[`renders <Header /> correctly 1`] = `
.c0 {
  display: block;
}

<header
  className="c0"
/>
`;

exports[`renders <Hr /> correctly 1`] = `
.c0 {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}

<hr
  className="c0"
/>
`;

exports[`renders <Img /> correctly 1`] = `
.c0 {
  border-style: none;
}

<img
  className="c0"
/>
`;

exports[`renders <Input /> correctly 1`] = `
.c0 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
}

.c0[type='checkbox'],
.c0[type='radio'] {
  box-sizing: border-box;
  padding: 0;
}

.c0[type='number']::-webkit-inner-spin-button,
.c0[type='number']::-webkit-outer-spin-button {
  height: auto;
}

.c0[type='search'] {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
  outline-offset: -0.125rem;
}

.c0[type='search']::-webkit-search-cancel-button,
.c0[type='search']::-webkit-search-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0::-webkit-file-upload-button {
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  font: inherit;
}

<input
  className="c0"
/>
`;

exports[`renders <Kbd /> correctly 1`] = `
.c0 {
  font-family: monospace,monospace;
  font-size: 1rem;
}

<kbd
  className="c0"
/>
`;

exports[`renders <Legend /> correctly 1`] = `
.c0 {
  box-sizing: border-box;
  color: inherit;
  display: table;
  max-width: 100%;
  padding: 0;
  white-space: normal;
}

<legend
  className="c0"
/>
`;

exports[`renders <Main /> correctly 1`] = `
.c0 {
  display: block;
}

<main
  className="c0"
/>
`;

exports[`renders <Mark /> correctly 1`] = `
.c0 {
  background-color: #ff0;
  color: #000;
}

<mark
  className="c0"
/>
`;

exports[`renders <Menu /> correctly 1`] = `
.c0 {
  display: block;
}

<menu
  className="c0"
/>
`;

exports[`renders <Nav /> correctly 1`] = `
.c0 {
  display: block;
}

<nav
  className="c0"
/>
`;

exports[`renders <Optgroup /> correctly 1`] = `
.c0 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
}

<optgroup
  className="c0"
/>
`;

exports[`renders <Pre /> correctly 1`] = `
.c0 {
  font-family: monospace,monospace;
  font-size: 1rem;
}

<pre
  className="c0"
/>
`;

exports[`renders <Progress /> correctly 1`] = `
.c0 {
  display: inline-block;
  vertical-align: baseline;
}

<progress
  className="c0"
/>
`;

exports[`renders <Samp /> correctly 1`] = `
.c0 {
  font-family: monospace,monospace;
  font-size: 1rem;
}

<samp
  className="c0"
/>
`;

exports[`renders <Section /> correctly 1`] = `
.c0 {
  display: block;
}

<section
  className="c0"
/>
`;

exports[`renders <Select /> correctly 1`] = `
.c0 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  text-transform: none;
}

<select
  className="c0"
/>
`;

exports[`renders <Small /> correctly 1`] = `
.c0 {
  font-size: 80%;
}

<small
  className="c0"
/>
`;

exports[`renders <Strong /> correctly 1`] = `
.c0 {
  font-weight: bolder;
}

<strong
  className="c0"
/>
`;

exports[`renders <Sub /> correctly 1`] = `
.c0 {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
  bottom: -0.25em;
}

<sub
  className="c0"
/>
`;

exports[`renders <Summary /> correctly 1`] = `
.c0 {
  display: list-item;
}

<summary
  className="c0"
/>
`;

exports[`renders <Sup /> correctly 1`] = `
.c0 {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
  top: -0.5em;
}

<sup
  className="c0"
/>
`;

exports[`renders <Svg /> correctly 1`] = `
.c0:not(:root) {
  overflow: hidden;
}

<svg
  className="c0"
/>
`;

exports[`renders <Textarea /> correctly 1`] = `
.c0 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: auto;
}

<textarea
  className="c0"
/>
`;

exports[`renders <Video /> correctly 1`] = `
.c0 {
  display: inline-block;
}

<video
  className="c0"
/>
`;
